src/Factories/FunFunFactory.php

```diff
@@ -598,7 +598,15 @@ public function newAddSubscriptionJsonPresenter(): AddSubscriptionJsonPresenter
 	}
 
 	public function newGetInTouchHtmlPresenter(): GetInTouchHtmlPresenter {
-		return new GetInTouchHtmlPresenter( $this->getLayoutTemplate( 'contact_form.html.twig' ), $this->getTranslator() );
+		return new GetInTouchHtmlPresenter(
+			$this->getLayoutTemplate( 'contact_form.html.twig' ),
+			$this->getTranslator(),
+			$this->getGetInTouchCategories()
+		);
+	}
+
+	public function getGetInTouchCategories(): array {
+		return json_decode( ( new SimpleFileFetcher() )->fetchFile( $this->getI18nDirectory() . '/data/contact_categories.json' ), true ) ?? [];
```

Comment:
Not sure about "masking" the error here by returning an empty array - the user will not be able to submit the contact form if there are no categories, which will be frustrating for her because she will probably have filled out the form. 

I think, from a UX perspective, it might be better to check if the categories array is not empty and throw a RuntimeException if it is. The user will see a nicely formatted error page (and can report to us that our error page is broken) and when https://phabricator.wikimedia.org/T163884 is implemented, we might even get notified.

Reply:
That's a good point. I've added logic to catch the exception thrown by `SimpleFileFetcher`, and I'd rather throw the `RuntimeException` one, instead of `FileFetchingException`, because `FileFetchingException` throws a default message in English, which looks strange in the middle of an error page in German. For this same reason `RuntimeException` is thrown with no message whatsoever. 
> we might even get notified.  
>
 :pray: :pray: